test/Microsoft.AspNetCore.Mvc.DataAnnotations.Test/Internal/DataAnnotationsMetadataProviderTest.cs

```diff
@@ -180,14 +194,21 @@ public void CreateDisplayMetadata_DisplayAttribute_NameFromResources()
 
             // Assert
             Assert.Equal("name from resources", context.DisplayMetadata.DisplayName());
+
+            if (stringLocalizerFactory != null)
```

Comment:
When you have to do bits like this in tests it typically means it should be a separate test :cry: